docs/flow/modin/experimental/xgboost.rst

```diff
@@ -0,0 +1,132 @@
+     This condition was chosen to combine  parallelization techniques: parallel actors and parallel threads.
+
+**Note**: ``num_actors`` parameter is made available for public functions :ref:`train <train>` and ``predict``
+method of :ref:`Booster<Booster>` class to allow fine-tuning for best performance in specific use cases.
+
+3. ``ray.util.placement_group`` is created to reserve all available Ray resources. After that
+   :ref:`ModinXGBoostActor <Actor>` objects are created using resources of previously created placement group.
+
+4. Data (`dtrain` for :ref:`_train <internal_train>`, `data` for :ref:`_predict <internal_predict>`) is split
+   between actors evenly. Internal function :ref:`_split_data_across_actors <split_data_across_actors>` runs
+   assigning row partitions to actors using internal function :ref:`_assign_row_partitions_to_actors <assign_parts>`.
+   This function creates dictionary in the form: `{actor_rank: ([part_i0, part_i3, ..], [0, 3, ..]), ..}` for training,
+   `{actor_rank: [part_i0, part_i1, ..], ..}` for prediction.
+
+**Note**: :ref:`_assign_row_partitions_to_actors <assign_parts>` takes into account IP addresses of row partitions
+of `dtrain` data to minimize excess data transfer.
+
+5. For each :ref:`ModinXGBoostActor <Actor>` object methods ``set_train_data`` or ``set_predict_data`` are
+   called remotely. Those methods run loading row partitions in actor according to dictionary with partitions
+   distribution from previous step. When data is passed in actor, auto-materialization of row partitions
+   (``ray.ObjectRef`` -> ``pandas.DataFrame``) happens.
+
+6. Methods ``train`` or ``predict`` of :ref:`ModinXGBoostActor <Actor>` class object are called remotely.
+
+   * `train`: method runs xgboost training on local data of actor, connects to ``Rabit Tracker`` for sharing
+     training state between actors and returns dictionary with `booster` and `evaluation results`.
+   * `predict`: method runs xgboost prediction on local data of actor and returns IP address of actor and partial
+     prediction (``pandas.DataFrame``).
+
+7. On the final stage results from actors are returned.
+
+   * `train`: `booster` and `evals_result` is returned using ``ray.get`` function from remote actor. Placement
+     group which was created on the step 3 is removed to free resources. :ref:`Booster<Booster>` object is
+     created and returned to user. 
+   * `predict`: using ``ray.wait`` function we wait until all actors finish computing local predictions. Placement group
+     which was created on the step 3 is removed to free resources. ``modin.pandas.DataFrame`` is created from
+     ``ray.ObjectRef`` objects which is returned from actors. Modin DataFrame is returned to user.
+
+
+Internal API
+''''''''''''
+.. _Actor:
+.. autoclass:: modin.experimental.xgboost.xgboost_ray.ModinXGBoostActor
+  :members:
+  :private-members:
```

Comment:
do we really want private methods to be documented here?..

Reply:
Yes, we want. In implementation details we refer on those internal functions.